sdk/storage/AzureStorageBlob/Source/Transport/URLSessionTransferManager+TransferDelegate.swift

```diff
@@ -91,9 +95,13 @@ extension URLSessionTransferManager: TransferDelegate {
 
     func transferDidComplete(_ transfer: Transfer) {
         guard let restorationId = (transfer as? TransferImpl)?.clientRestorationId else { return }
-        guard let delegate = client(forRestorationId: restorationId) as? TransferDelegate else { return }
+        let delegate = client(forRestorationId: restorationId) as? TransferDelegate
+        guard delegate != nil || (transfer as? BlobTransfer)?.progressHandler != nil else { return }
         DispatchQueue.main.async {
-            delegate.transferDidComplete(transfer)
+            delegate?.transferDidComplete(transfer)
+            if let blobTransfer = transfer as? BlobTransfer {
+                blobTransfer.progressHandler?(blobTransfer)
```

Comment:
I think we need to do the same thing in `self.transfer(_:didFailWithError:)`.

Reply:
Done.